### ISSUE TITLE
All tests fail due to "Object/factory with  type ConvenientTestSlot is already registered inside GetIt."

### ISSUE THREAD

**Issue by ronba** [fzyzcjy/flutter_convenient_test#271]
**Describe the bug**
When starting the test suite the following exception is always thrown:

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following ArgumentError was thrown running a test:
Invalid argument(s): Object/factory with  type ConvenientTestSlot is already registered inside
GetIt.
```

**To Reproduce**
1. Create a main_test.dart file which looks something like this:
```dart
import 'package:convenient_test_dev/convenient_test_dev.dart';
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  convenientTestMain(MyConvenientTestSlot(), () {
    group('simple test group', () {
      tTestWidgets('choose some fruits', (t) async {
        expect(true, true);
      });
    });
  });
}

class MyConvenientTestSlot extends ConvenientTestSlot {
  @override
  Future<void> appMain(AppMainExecuteMode mode) async => main();

  @override
  BuildContext? getNavContext(ConvenientTest t) =>
      StoriesRouter.navigatorKey.currentContext;
}
```
2) Start the test suite.

**Expected behavior**
The first test should pass.

**Actual behavior**
The test fails with the following exception:

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following ArgumentError was thrown running a test:
Invalid argument(s): Object/factory with  type ConvenientTestSlot is already registered inside
GetIt.

When the exception was thrown, this was the stack:
#0      throwIf (package:get_it/get_it_impl.dart:7:18)
#1      _GetItImplementation._register (package:get_it/get_it_impl.dart:803:5)
#2      _GetItImplementation.registerSingleton (package:get_it/get_it_impl.dart:587:5)
#3      convenientTestMain (package:convenient_test_dev/src/functions/core.dart:66:11)
#4      main (file:///Users/ronba/dev/projects/comobi/comobi/app/integration_test/main_test.dart:7:3)
#5      MyConvenientTestSlot.appMain (file:///Users/ronba/dev/projects/comobi/comobi/app/integration_test/main_test.dart:28:58)
#6      tTestWidgets.<anonymous closure>.<anonymous closure> (package:convenient_test_dev/src/functions/core.dart:204:12)
#7      ConvenientTest.withActiveInstance (package:convenient_test_dev/src/functions/core.dart:46:17)
#8      tTestWidgets.<anonymous closure> (package:convenient_test_dev/src/functions/core.dart:200:30)
#9      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:171:29)
```

**Desktop (please complete the following information):**
 - MacOS 
 - Ventura 13.0

**Smartphone (please complete the following information):**
 - Device: Android Emulator
 - OS: 12

**Additional context**
I've reproduced this issue with the master branch.

As a workaround, I locally applied a small hack as follows:
in the [core.dart file, convenientTestMain](https://github.com/fzyzcjy/flutter_convenient_test/blob/23a6a1c6c709357d74270e379f78d0c993d6beec/packages/convenient_test_dev/lib/src/functions/core.dart#L53):

```dart
  if (myGetIt.isRegistered<ConvenientTestSlot>()) {
    myGetIt.unregister<ConvenientTestSlot>();
  }

  if (myGetIt.isRegistered<ConvenientTestManagerClient>()) {
    myGetIt.unregister<ConvenientTestManagerClient>();
  }
```

**Comment by ronba** [comment]
Closing. Complete PEBCAK.

The issue was so obvious it's painful. In the sample code I sent:
```
  Future<void> appMain(AppMainExecuteMode mode) async => main();
```
I originally imported the file which had the app's main method.  
Then I created the sample above - which also had a main method...   
The next time I saved - it removed the import with the app's main method and now appMain was pointing to the test's main 🤦 .
